includes/login-register.php

```diff
@@ -159,6 +159,73 @@ function edd_log_user_in( $user_id, $user_login, $user_pass, $remember = false )
 
 }
 
+add_filter( 'wp_login_errors', 'edd_login_register_error_message', 10, 2 );
+/**
+ * Changes the WordPress login confirmation message when using EDD's reset password link.
+ *
+ * @since 2.10
+ * @param object \WP_Error $errors
+ * @param string $redirect
+ * @return void
+ */
+function edd_login_register_error_message( $errors, $redirect ) {
+	$action_is_confirm   = 'confirm' === filter_input( INPUT_GET, 'checkemail', FILTER_SANITIZE_STRING );
+	$edd_action_is_reset = 'confirm' === filter_input( INPUT_GET, 'edd_reset_password', FILTER_SANITIZE_STRING );
+	$redirect_url        = filter_input( INPUT_GET, 'edd_redirect', FILTER_SANITIZE_URL );
+	if ( $action_is_confirm && $edd_action_is_reset && $redirect_url ) {
+		$errors->remove( 'confirm' );
+		$errors->add(
+			'confirm',
+			apply_filters(
+				'edd_login_register_error_message',
+				sprintf(
+					/* translators: %s: Link to the referring page. */
+					__( 'Follow the instructions in the confirmation email you just received, then <a href="%s">return to what you were doing</a>.', 'easy-digital-downloads' ),
+					esc_url( $redirect_url )
+				),
+				$redirect_url
+			),
+			'message'
+		);
+	}
+
+	return $errors;
+}
+
+/**
+ * Gets the lost password URL, customized for EDD. Using this allows the password
+ * reset form to redirect to the login screen with the EDD custom confirmation message.
+ *
+ * @since 2.10
+ * @return string
+ */
+function edd_get_lostpassword_url() {
+	$url = false;
+	// `wp_get_raw_referer` was added in WP 4.5; this check is required as we still support 4.4.
+	if ( function_exists( 'wp_get_raw_referer' ) ) {
+		$url = wp_get_raw_referer();
+	} else {
+		if ( ! empty( $_REQUEST['_wp_http_referer'] ) ) {
+			$url = wp_unslash( $_REQUEST['_wp_http_referer'] );
+		} elseif ( ! empty( $_SERVER['HTTP_REFERER'] ) ) {
+			$url = wp_unslash( $_SERVER['HTTP_REFERER'] );
+		}
+	}
+	if ( empty( $url ) ) {
+		$url = edd_get_checkout_uri();
```

Comment:
I just saw an issue come up with the auto register extension which pointed out that there is a setting for the login redirect. I think that should be the fallback, instead of the checkout page.